src/app/zap-templates/common/ClusterTestGeneration.js

```diff
@@ -609,6 +609,24 @@ function octetStringEscapedForCLiteral(value)
   });
 }
 
+// Structs may not always provide values for optional members.
+function if_include_struct_item_value(structValue, name, options)
+{
+  let hasValue = (name in structValue);
+  if (hasValue) {
+    return options.fn(this);
+  }
+
+  if (!this.isOptional) {
+    if (!hasValue) {
```

Comment:
That seems always `true` since there is a previous `if (hasValue)` check  and I would expect `hasValue` to be always `false` ?

Reply:
Good catch!  This code used to look a bit different, then got refactored and you are right that the new code makes no sense.